torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -869,11 +865,27 @@ std::vector<int64_t> TensorDomain::mapDomainCtoP(
       continue;
     }
 
-    consumer_to_producer[itc] = itp;
+    // consumer_to_producer[itc] = itp;
+    dom_map.emplace_back(std::make_pair(itp, itc));
     itc++;
     itp++;
   }
-  return consumer_to_producer;
+  return dom_map;
+}
+
+std::vector<std::pair<IterDomain*, IterDomain*>> TensorDomain::mapRootPandC(
+    const TensorDomain* producer,
+    const TensorDomain* consumer) {
+  auto consumer_root = consumer->rootDomain();
+  auto producer_root = producer->hasRFactor() ? producer->rfactorDomain()
+                                              : producer->rootDomain();
+  std::vector<std::pair<IterDomain*, IterDomain*>> root_id_map;
+  for (const auto& m : mapDomainPandC(producer_root, consumer_root)) {
+    auto producer_axis = producer_root[m.first];
+    auto consumer_axis = consumer_root[m.second];
+    root_id_map.emplace_back(std::make_pair(producer_axis, consumer_axis));
+  }
+  return root_id_map;
 }
 
 // Create a map from consumer root IterDomains -> producer root IterDomains.
```

Comment:
Update this comment please.

Reply:
I just deleted it as it's redundant to have the same comment both in the header and source files.